src/boot/efi/stub.c

```diff
@@ -76,10 +75,17 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
                 cmdline = line;
 
 #if ENABLE_TPM
+        } else if (cmdline_len > 0) {
+                UINTN i;
+                options = AllocatePool(cmdline_len * sizeof(CHAR16));
+                for (i= 0; i < cmdline_len; i++)
+                        options[i] = cmdline[0];
```

Comment:
Shouldn't you be accessing `cmdline[i]` here?

Reply:
yes......

How does this even work!!!!!!!!

Reply:
please use stra_to_str() for this conversion, it's a lot more correct